vortexor/src/load_balancer.rs

```diff
@@ -0,0 +1,206 @@
+//! Module responsible for interacting with RPC services.
+//! Adapted from jito-replayer code.
+
+use {
+    crossbeam_channel::{Receiver, RecvTimeoutError, Sender},
+    dashmap::DashMap,
+    log::{error, info},
+    solana_client::{pubsub_client::PubsubClient, rpc_client::RpcClient},
+    solana_metrics::{datapoint_error, datapoint_info},
+    solana_sdk::{
+        clock::Slot,
+        commitment_config::{CommitmentConfig, CommitmentLevel},
+    },
+    std::{
+        sync::{
+            atomic::{AtomicBool, AtomicU64, Ordering},
+            Arc,
+        },
+        thread::{self, sleep, Builder, JoinHandle},
+        time::{Duration, Instant},
+    },
+};
+
+/// The RpcLoadBalancer can support providing a RpcClient which has the most
+/// up-to-date slot information among a set of configured RPC servers.
+/// This service starts threads which will subscribe to the slot events
+/// using the corresponding web socket for a RPC server. This mechansim can
+/// load balance the RPC calls to multiple RPC servers.
+pub struct RpcLoadBalancer {
+    /// (ws_url, slot)
+    server_to_slot: Arc<DashMap<String, Slot>>,
```

Comment:
`DashMap` is not the right structure here - every thread knows exactly what space it should be inserting into and the "consumer" just iterates this, not search by hash.

We should use a `Vec<Arc<(String, AtomicSlot)>>` or something similar.

Reply:
Will do